internal/files/files.go

```diff
@@ -0,0 +1,90 @@
+package files
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/Arti9991/shortener/internal/logger"
+	"github.com/Arti9991/shortener/internal/storage"
+	"go.uber.org/zap"
+)
+
+type FileStor struct {
+	ID       int    `json:"uuid"`
+	Shorturl string `json:"short_url"`
+	Origurl  string `json:"original_url"`
+}
+type FileData struct {
+	ID   int
+	stor *storage.Data
+	path string
+}
+
+func NewFiles(path string, stor *storage.Data) *FileData {
```

Comment:
```suggestion
func NewFiles(path string, stor *storage.Data) (*FileData, error) {
```
Нужны, как минимум проверки на имя и доступность файла. Возможно, на возможность его создания. В настоящий момент если указать `-f=""` будет постоянная ругань в логах на невозможность работы с файлом.
В этом случае правильный паттерн - вообще откатиться на in-memory хранилище.

Reply:
Обновил функцию инициализации. Добавил проброс ошибки для нее на верхний уровень. Также добавил флаг "inMemory" в структуру. При инициализации пробую открыть файл на чтение-запись, если ошибка (или путь пустой), то флаг ставится true и программа работает только с временной памятью